File: ljh/javascript/level1/test/SortMyWay.test.js
import SortMyWay from '../SortMyWay';
test(`문자열 내 마음대로 정렬하기`, () => {
    expect(new SortMyWay().solution(["sun", "bed", "car"], 1)).toEqual(expect.arrayContaining(["car", "bed", "sun"]));
    expect(new SortMyWay().solution(["abce", "abcd", "cdx"], 2)).toEqual(expect.arrayContaining(["abcd", "abce", "cdx"]));
});

File: ljh/java/src/level1/test/SortMyWayTest.java
package level1.test;

import static org.junit.Assert.assertArrayEquals;

import java.util.Arrays;

import org.junit.Before;
import org.junit.Test;

import level1.SortMyWay;

public class SortMyWayTest {
	private String[][] strings; // 입력 string
	private String[][] answers; // 출력
	private int[] ns; // 입력 n 
	private SortMyWay sortMyWay;
	
	@Before
	public void setUp() throws Exception{
		strings = new String[][] {
			{"sun", "bed", "car"},
			{"abce", "abcd", "cdx"}
		};
		answers = new String[][] {
			{"car", "bed", "sun"},
			{"abcd", "abce", "cdx"}
		};
		ns = new int[] {1, 2};		
	}
	
	@Test
	public void test() {
		int index = 0;
		int range = strings.length;
		String[] result = null;
		
		sortMyWay = new SortMyWay();
		
		do {
			result = sortMyWay.solution(strings[index], ns[index]);
			assertArrayEquals(String.format("answer: %s, result: %s", Arrays.toString(answers[index]), Arrays.toString(result)), result, answers[index]);
		} while((++index) < range);
		
	}
}
